### ISSUE TITLE
nsepy not working

### ISSUE THREAD

**Issue by adityapmehta** [swapniljariwala/nsepy#109]
when i run my program shows building dictionary.... but does not go ahead.

**Comment by ankit7070** [comment]
NSE seems to have blocked automated HTTP based scraping since this monday.

It works only after 11AM IST now.

**Comment by adityapmehta** [comment]
11 am till what time ? any idea

**Comment by imd2078** [comment]
Any idea when nsepy will work?

**Comment by Piyush-Bijwal** [comment]
any luck on this??? Could anyone share any alternative way of getting NSE data?

**Comment by akshayn83** [comment]
Not working at 23:46 IST either. any other way around this?

**Comment by RanjitV17** [comment]
Works only between 12:00PM and 9PM IST.

**Comment by Bogaso** [comment]
I noticed NSE scaping work only from 12.00pm to 11pm (approx). So you will not be able to scrape during other time, by whatever software/program you use. This is ridiculous.  I have already reported this issue in some other thread here, and interested to get any workaround.

**Comment by Prasanth777** [comment]
Any alternatives available to fetch historical data as of now? 
Will NSE enable HTTP scrapping ever again?

**Comment by omarrohit20** [comment]
Need to fix the code for url. Unless fix not available in nsepy temporarily fix locally in urls.py by replacing hostname from www.nseindia.com to https://nseindia.com . To confirm fix check following curl request same time when nsepy is not working. `curl 'https://nseindia.com/products/dynaContent/common/productsSymbolMapping.jsp?fromDate=23-10-2010&segmentLink=3&series=EQ&toDate=02-03-2011&symbolCount=2&symbol=ACC&dataType=PRICEVOLUMEDELIVERABLE&dateRange=' -H 'Connection: keep-alive' -H 'User-Agent: Chrome/79.0.3945.88' -H 'Referer: https://www.nseindia.com'`

**Comment by RanjitV17** [comment]
@omarrohit20 : Your curl request closed with this error "curl: (56) OpenSSL SSL_read: SSL_ERROR_SYSCALL, errno 60". Works for you?

**Comment by Praxal** [comment]
@omarrohit20 : 
Can you provide your copy of 'urls.py' ? 
It's still not working on my end. However, curl request response was good.

**Comment by omarrohit20** [comment]
SSL_ERROR_SYSCALL I was seeing due to 'User-Agent: python-requests/2.22.0'. Replacing the same for chrome worked fine. Seems NSE blocking due to multiple reason. Always check with curl which host and user agent is working make modifications accordingly in urls.py. Till now I saw following hostname 'www.nseindia.com', 'https://www.nseindia.com' and 'https://nseindia.com'

**Comment by yogiboy** [comment]
It seems, they will be blocking free programmatic access to data soon 🤨

**Comment by sf99167** [comment]
I have tried after 12 pm in the afternoons on weekdays.. no luck at a all. I get empty datasets in my results and in my other program, it just crashes like there is no internet access. I guess the access to NSE has been blocked. Any alternative methods or sites that can be used to access data??

**Comment by monk-boop** [comment]
Yup. It looks like it's blocked. I've been trying for a few days now.
